src/Microsoft.ML.AutoML/Utils/UserInputValidationUtil.cs

```diff
@@ -183,14 +183,19 @@ private static void ValidateValidationData(IDataView trainData, IDataView valida
 
             const string schemaMismatchError = "Training data and validation data schemas do not match.";
 
-            if (trainData.Schema.Count != validationData.Schema.Count)
+            if (trainData.Schema.Count(c => !c.IsHidden) != validationData.Schema.Count(c => !c.IsHidden))
             {
                 throw new ArgumentException($"{schemaMismatchError} Train data has '{trainData.Schema.Count}' columns," +
                     $"and validation data has '{validationData.Schema.Count}' columns.", nameof(validationData));
             }
 
             foreach (var trainCol in trainData.Schema)
```

Comment:
Could use a comment...
```suggestion
            // Also indirectly checks for new columns in the validation datasets as we above enforce the column counts are equal
            foreach (var trainCol in trainData.Schema)
```

I was otherwise going to suggest we check the reverse direction. A comment helps future readers to not have to think through it.